pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -461,6 +468,10 @@ func isCustomTask(ctx context.Context, rprt ResolvedPipelineRunTask) bool {
 	return cfg.FeatureFlags.EnableCustomTasks && !invalidSpec && (isTaskRefCustomTask || isTaskSpecCustomTask)
 }
 
+func isMatrixed(rprt ResolvedPipelineRunTask) bool {
```

Comment:
I might suggest not making this a separate method, since it can be confusing to see an `isMatrixed(rprt)` method and a `rprt.IsMatrixed()` method, and have them not be the same thing.

It took me a minute to realize `isMatrixed` here is only used to _populate_ `rprt.Matrixed` below, which is read out by `rprt.IsMatrixed()`.